### PR TITLE
 - fixed validation-guide

### PR DESCRIPTION
**Pull request by Z3NTL3**
## Description

The old validation guide hosted on the docs was a bit outdated. It was good but, it was only for a single object type validation. Instead with this PR I have changed this. Now the validator can handle any type of object and validate within ease.

I have also made easy examples about how to make custom error message format when a context does not fit our validation scheme.

## Type of change
- Improved validation guide
- Custom error message format for validator (near choice)
- A function capable of handling all type of objects data to validate against
- Improvements towards the debug tests and output showcases
- Easy showcase about how to make an custom tag format validation

## Checklist:
- I have alot of comments in my code
- Well formated OUTPUT console to provide a better way of understanding
- Easy and adaptable code

### CODE REVIEW

File: docs/guide/validation.md [hunk 1]
```diff
@@ -15,69 +15,153 @@ You can find the detailed descriptions of the _validations_ used in the fields c
...
+	"fmt"
+	"log"
+	"strings"
 
-type ErrorResponse struct {
-    FailedField string
-    Tag         string
-    Value       string
-}
+	"github.com/go-playground/validator"
```
Comment (efectn) [inline docs/guide/validation.md:25]: can you change this one with v10

File: docs/guide/validation.md [hunk 2]
```diff
@@ -15,69 +15,153 @@ You can find the detailed descriptions of the _validations_ used in the fields c
...
+		Message string 	`json:"message"`
+	}
+)
+
+// This is the validator instance
+// for more information see: https://github.com/go-playground/validator
+var validate = validator.New()
+
+func(v XValidator) Validate(data interface{}) []ErrorResponse{
+	error_ := []ErrorResponse{}
```
Comment (efectn) [inline docs/guide/validation.md:57]: errors would be better

File: docs/guide/validation.md [hunk 3]
```diff
@@ -15,69 +15,153 @@ You can find the detailed descriptions of the _validations_ used in the fields c
...
+		}
+	}
+
+	return error_
+}
 
-    //Return user
-   return c.JSON(user)
+func main() {
+	Validator_ := &XValidator{
```
Comment (efectn) [inline docs/guide/validation.md:77]: Whu did you add _ to end of the variable. Not sure it's good variable naming

**Review by efectn — COMMENTED** [review]
can you also check indentions

**Comment by Z3NTL3** [comment]
> can you also check indentions

fixed it, and yea i though one of the modules i used already did have that name so i simply added ``_`` but i see that i were wrong

**Comment by ReneWerner87** [comment]
@Z3NTL3 pls check again the whole code for the in indentions
image
image
image

**Comment by Z3NTL3** [comment]
Hello,

I have formatted everything with ``go fmt``, check the commit. All indentations are correct now.

**Review by ReneWerner87 — APPROVED** [review]

File: docs/guide/validation.md (hunk 1 above)
Comment (ReneWerner87) [reply docs/guide/validation.md:26]: @Z3NTL3

Comment (renanbastos93) [reply docs/guide/validation.md:26]: nice catch @efectn

Comment (Z3NTL3) [reply docs/guide/validation.md]: Fixed it, look at <a href="https://github.com/gofiber/fiber/pull/2517/commits/34cffe582891693b83347708f07c6f7fb601edd3">commit</a>

**Review by Z3NTL3 — COMMENTED** [review]
Updated validator pkg import to v10

**Review by renanbastos93 — APPROVED** [review]

File: docs/guide/validation.md [hunk 4]
```diff
@@ -8,76 +8,162 @@ sidebar_position: 5
...
-type User struct{
-    Name          string  `validate:"required,min=3,max=32"`
-    // use `*bool` here otherwise the validation will fail for `false` values
-    // Ref: https://github.com/go-playground/validator/issues/319#issuecomment-339222389
-    IsActive      *bool   `validate:"required"`
-    Email         string  `validate:"required,email,min=6,max=32"`
-    Job           Job     `validate:"dive"`
-}
+func (v XValidator) Validate(data interface{}) []ErrorResponse {
+	Error := []ErrorResponse{}
```
Comment (efectn) [inline docs/guide/validation.md:57]: Can you make this var's name validationErrors

File: docs/guide/validation.md [hunk 5]
```diff
@@ -8,76 +8,162 @@ sidebar_position: 5
...
-            var element ErrorResponse
-            element.FailedField = err.StructNamespace()
-            element.Tag = err.Tag()
-            element.Value = err.Param()
-            errors = append(errors, &element)
-        }
-    }
-    return errors
+func main() {
+	Validator := &XValidator{
```
Comment (efectn) [inline docs/guide/validation.md:78]: This one can be unexported too

Comment (Z3NTL3) [reply docs/guide/validation.md]: Oke, completed 👍

File: docs/guide/validation.md (hunk 4 above)
Comment (Z3NTL3) [reply docs/guide/validation.md]: Oke, completed 👍

**Review by efectn — APPROVED** [review]

**Pull request merged**